dev/tools/localization/gen_l10n.dart

```diff
@@ -473,13 +467,76 @@ Future<void> main(List<String> arguments) async {
     'pub',
     'run',
     'intl_translation:generate_from_arb',
-    '--output-dir=${l10nDirectory.path}',
+    '--output-dir=${localizationsGenerator.l10nDirectory.path}',
     '--no-use-deferred-loading',
-    outputFile.path,
+    localizationsGenerator.outputFile.path,
     ...arbFilenames,
   ]);
   if (generateFromArbResult.exitCode != 0) {
     stderr.write(generateFromArbResult.stderr);
     exit(1);
   }
 }
+
+class LocalizationsGenerator {
+  LocalizationsGenerator(
```

Comment:
I would tweak this slightly by separating the dependencies from the data. Provide `fs` in the constructor, and then create a method that populates or returns outputFile, templateArbFile, et cetera